### ISSUE TITLE
[BUG] Transpose bug in `reward2go` when the last dim is not 1

### ISSUE THREAD

**Issue by SamAdamDay** [pytorch/rl#2086]
## Describe the bug

When calling `reward2go` with a shape whose last dimension is not 1, the results are incorrect due to reshaping rather than transposing.

## To Reproduce

Minimal example:

```python
from torchrl.objectives.value.functional import reward2go
import torch
reward = torch.zeros(4, 2)
reward[3, 0] = 1
reward[3, 1] = -1
done = torch.zeros(4, 2, dtype=bool)
done[3, :] = True
reward2go(reward, done, 0.9)
```

Output:

```python
tensor([[ 0.7290,  0.8100],
        [ 0.9000,  1.0000],
        [-0.7290, -0.8100],
        [-0.9000, -1.0000]])
```

## Expected behavior

```python
tensor([[ 0.7290, -0.7290],
        [ 0.8100, -0.8100],
        [ 0.9000, -0.9000],
        [ 1.0000, -1.0000]])
```

## System info

```python
import torchrl, numpy, sys
print(torchrl.__version__, numpy.__version__, sys.version, sys.platform)
```

```
0.4.0+8570bd3 1.26.2 3.11.6 (main, Oct  8 2023, 05:06:43) [GCC 13.2.0] linux
```

## Reason and Possible fixes

In the last step of `reward2go` the cumsum is reshaped:

```python
    if cumsum.shape != shape:
        cumsum = cumsum.view(shape)
```

It should be:

```python
    cumsum = cumsum.transpose(-2, -1)
```

to undo an earlier transpose.

I'm happy to contribute a PR with a fix and an extra test.

## Checklist

- [X] I have checked that there is no similar issue in the repo (**required**)
- [X] I have read the [documentation](https://github.com/pytorch/rl/tree/main/docs/) (**required**)
- [X] I have provided a minimal working example to reproduce the bug (**required**)
